config/src/main/java/org/springframework/security/config/annotation/authentication/builders/AuthenticationManagerBuilder.java

```diff
@@ -149,17 +150,47 @@ public InMemoryUserDetailsManagerConfigurer<AuthenticationManagerBuilder> inMemo
 	 * {@link #getDefaultUserDetailsService()} method. Note that additional
 	 * {@link UserDetailsService}'s may override this {@link UserDetailsService} as the
 	 * default. See the <a href=
-	 * "https://docs.spring.io/spring-security/site/docs/current/reference/htmlsingle/#user-schema"
+	 * "https://docs.spring.io/spring-security/reference/servlet/appendix/database-schema.html"
 	 * >User Schema</a> section of the reference for the default schema.
 	 * </p>
 	 * @return a {@link JdbcUserDetailsManagerConfigurer} to allow customization of the
 	 * JDBC authentication
 	 * @throws Exception if an error occurs when adding the JDBC authentication
+	 * @deprecated Use jdbcAuthenticationWithPasswordManagement instead, as this keeps the password encoding up to
+	 * date over time. Please consult the migrationn documentation as database changes might be necessary.
 	 */
 	public JdbcUserDetailsManagerConfigurer<AuthenticationManagerBuilder> jdbcAuthentication() throws Exception {
 		return apply(new JdbcUserDetailsManagerConfigurer<>());
 	}
 
+	/**
+	 * Add JDBC authentication to the {@link AuthenticationManagerBuilder} and return a
+	 * {@link JdbcUserPasswordDetailsManagerConfigurer} to allow customization of the JDBC
+	 * authentication. This manager will automatically ensure that passwords are encooded
+	 * with the configured password encoder when a user logs in.
+	 *
+	 * <p>
+	 * When using with a persistent data store, it is best to add users external of
+	 * configuration using something like <a href="https://flywaydb.org/">Flyway</a> or
+	 * <a href="https://www.liquibase.org/">Liquibase</a> to create the schema and adding
+	 * users to ensure these steps are only done once and that the optimal SQL is used.
+	 * </p>
+	 *
+	 * <p>
+	 * This method also ensure that a {@link UserDetailsService} is available for the
+	 * {@link #getDefaultUserDetailsService()} method. Note that additional
+	 * {@link UserDetailsService}'s may override this {@link UserDetailsService} as the
+	 * default. See the <a href=
+	 * "https://docs.spring.io/spring-security/reference/servlet/appendix/database-schema.html"
+	 * >User Schema</a> section of the reference for the default schema.
+	 * </p>
+	 * @return a {@link JdbcUserDetailsManagerConfigurer} to allow customization of the
+	 * JDBC authentication
+	 * @throws Exception if an error occurs when adding the JDBC authentication
+	 */
+	public JdbcUserPasswordDetailsManagerConfigurer<AuthenticationManagerBuilder> jdbcAuthenticationWithPasswordManagement() throws Exception {
```

Comment:
If possible, let's hold off on this addition since it's preferred that applications publish a `UserDetailsService` bean instead.

Reply:
Sure.

Does this mean that I should strip the XML-based configuration, too?

Reply:
Good question.

No, XML is a different configuration type. In Java, there are two configuration types for the `AuthenticationManager` and we are steering folks to the bean-based one instead of the DSL one.

Reply:
Cool. I suspect that AuthenticationManagerBuilder.jdbcAuthentication should be deprecated, then?

Reply:
Added the deprecation. Only thing that makes sense to me when the underlying class disappears.